File: mobile/src/main/java/at/shockbytes/remote/network/model/FileTransferResponse.java
package at.shockbytes.remote.network.model;

/**
 * @author Martin Macheiner
 *         Date: 26.09.2017.
 */

public class FileTransferResponse {

    private final String filename;
    private final byte[] content;
    private final String exception;

    public FileTransferResponse() {
        this(null, null, null);
    }

    public FileTransferResponse(String filename, byte[] content, String exception) {
        this.filename = filename;
        this.content = content;
        this.exception = exception;
    }

    public String getFilename() {
        return filename;
    }

    public byte[] getContent() {
        return content;
    }

    public String getException() {
        return exception;
    }

    public boolean isEmpty() {
        return content == null;
    }

    @Override
    public String toString() {
        int size = content == null ? -1 : content.length;
        return "Filename: " + filename + "\nException: " + exception + "\nSize: " + size;
    }
}

File: mobile/src/main/java/at/shockbytes/remote/network/model/FileTransferResponse.kt
package at.shockbytes.remote.network.model

import java.util.*

/**
 * @author Martin Macheiner
 * Date: 26.09.2017.
 */

data class FileTransferResponse (val filename: String = "",
                                 private val transferCode: Int = TransferCode.PARSE_ERROR.ordinal,
                                 val content: ByteArray?,
                                 private val exception: String?) {

    enum class TransferCode {
        OKAY, NO_ACCESS, TOO_BIG, ENCODING_ERROR, PARSE_ERROR
    }

    val isEmpty: Boolean
        get() = content?.isEmpty() ?: false

    fun getTransferCodeAsEnum() : TransferCode {
        return TransferCode.values()[transferCode]
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FileTransferResponse

        if (filename != other.filename) return false
        if (!Arrays.equals(content, other.content)) return false
        if (transferCode != other.transferCode) return false
        if (exception != other.exception) return false

        return true
    }

    override fun hashCode(): Int {
        var result = filename.hashCode()
        result = 31 * result + Arrays.hashCode(content)
        result = 31 * result + transferCode
        result = 31 * result + (exception?.hashCode() ?: 0)
        return result
    }

}
